### DISCUSSION TITLE
Why is z-index:0 applied to all divs?

### DISCUSSION THREAD

**Discussion by nikkwong** [necolas/react-native-web#2547] [Q&A]
This makes it impossible for divs to breakout of their stacking context, and implicitly ties the hierarchy of the DOM tree to the stacking order with no way to override (typically what z-index is used for). I'm trying to add some tooltips using portals and whatnot but the heavy-handed application of `z-index: 0` everywhere makes this somewhat impossible. Any thoughts/workarounds?

**Comment by itsramiel** [comment]
I am also looking for a similar solution. I wanted to use floating ui, but the stacking context makes it hard. The reason it was added is to make it inline with react native on mobile.

One solution I had is to patch that change 🤔

@nikkwong did you come up with something?
@necolas What is the suggested approach when it comes to adding tooltips and popovers?

**Comment by Fausto95** [comment]
I was facing the same issue using `@floating-ui`.

The solution that worked for me is using a portal, I'm using `@gorhom/portal` for that and it also works with `react-dom`'s createPortal on the web @itsramiel!

**Comment by nikkwong** [comment]
Not a good solution, a year later. Portals have their own problems, like not being able to break out of RN's `<Modal>`. Hoping for an actual solution here.

**Comment by ROMUALDO-TXT** [comment]
I'm having problems with it right now, considering configuring every View style manually to make my components work.

**Comment by hadrien-xbto** [comment]
A simple fix for that is to reset the specific change. Life saver for me when I applied that CSS, I manage to remove all my CSS zIndex hacks nightmares :
```
<style>
  [class^="css-view-"] {
      z-index: auto;
  }
</style>
```

Probably not a bulletproof solution, but if it can help some...

**Comment by kaspernj** [comment]
I tested on Android and it didn't seem to be a problem there, so maybe it should be fixed in the React Native Web project?
